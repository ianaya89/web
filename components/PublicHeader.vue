<template>
  <header class="header--public flex w-full h-24 justify-between">
    <div class="flex">
      <ul class="flex">
        <li v-for="(item, index) in leftNav" :key="index">
          <nuxt-link :to="item.href" :class="{'text-red': item.active}">
            {{ item.label }}
          </nuxt-link>
        </li>
      </ul>
    </div>
    <div class="flex">
      <ul class="flex justify-end">
        <li>
          <nuxt-link v-if="!isAuthenticated" :to="localePath({name: 'login'})" class="my-auto text-white">
            {{ $t('public.header.login') }}
          </nuxt-link>
          <nuxt-link v-else :to="resolvedEnterInApp" class="my-auto text-white">
            {{ fullName }}
          </nuxt-link>
        </li>
      </ul>
    </div>
  </header>
</template>

<script>
import { mapGetters } from 'vuex'
import { MeQuery } from '@/services/graphql'

export default {
  name: 'PublicHeader',

  data: () => ({
    me: null
  }),

  apollo: {
    me: {
      ...MeQuery,
      skip() {
        return !this.isAuthenticated
      }
    }
  },

  computed: {
    ...mapGetters({
      isAuthenticated: 'auth/isAuthenticated'
    }),

    leftNav() {
      return [
        {
          label: this.$t('public.header.teams'),
          href: this.localePath({ name: 'index' })
        },
        {
          label: this.$t('public.header.brands'),
          href: this.localePath({ name: 'brand' })
        },
        {
          label: this.$t('public.header.news'),
          href: this.localePath({ name: 'news' }),
          active: true
        }
      ]
    },

    resolvedEnterInApp() {
      if (this.me !== null) {
        const unitId = this.$get(this.me, 'units.0.id')
        return this.localePath({
          name: 'unit-id-dashboard',
          params: { id: unitId }
        })
      } else {
        return {}
      }
    },

    fullName() {
      if (this.me !== null) {
        const firstName = this.$get(this.me, 'profile.firstName', '-')
        const lastName = this.$get(this.me, 'profile.lastName', '-')
        return `${firstName} ${lastName}`
      } else {
        return ''
      }
    }
  }
}
</script>

<style lang="postcss">
.header--public {
  ul {
    @apply .items-center;
  }

  li {
    @apply .flex;
  }

  a {
    @apply .text-menu .px-8 .py-4 .text-grey-l2 .font-medium;

    &:hover {
      @apply .text-black;
    }
  }
}
</style>
