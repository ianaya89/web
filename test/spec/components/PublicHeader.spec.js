import { shallowMount } from '@vue/test-utils'
import PublicHeader from '@/components/PublicHeader'

describe('PublicHeader', () => {
  it('renders PublicHeader', () => {
    const wrapper = shallowMount(PublicHeader, {
      mocks: {
        $store: {
          getters: {
            'auth/isAuthenticated': false
          }
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('returns current logged in user full name', () => {
    const wrapper = shallowMount(PublicHeader, {
      mocks: {
        $store: {
          getters: {
            'auth/isAuthenticated': false
          }
        }
      }
    })

    expect(wrapper.vm.fullName).toEqual('')

    wrapper.vm.me = {
      profile: {
        firstName: 'John',
        lastName: 'Doe'
      }
    }

    expect(wrapper.vm.fullName).toEqual('John Doe')
  })

  it('returns link to dashboard for logged in user', () => {
    const wrapper = shallowMount(PublicHeader, {
      mocks: {
        $store: {
          getters: {
            'auth/isAuthenticated': true
          }
        }
      }
    })

    expect(wrapper.vm.resolvedEnterInApp).toEqual({})

    wrapper.vm.me = {
      units: [
        {
          id: 1
        }
      ]
    }

    expect(wrapper.vm.resolvedEnterInApp).toEqual({
      name: 'unit-id-dashboard',
      params: { id: 1 }
    })
  })
})
