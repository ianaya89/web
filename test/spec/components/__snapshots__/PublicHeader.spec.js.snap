// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicHeader renders PublicHeader 1`] = `
<header
  class="header--public flex w-full h-24 justify-between"
>
  <div
    class="flex"
  >
    <ul
      class="flex"
    >
      <li>
        <a
          class=""
        >
          
          TRANSLATE:public.header.teams
        
        </a>
      </li>
      <li>
        <a
          class=""
        >
          
          TRANSLATE:public.header.brands
        
        </a>
      </li>
      <li>
        <a
          class="text-red"
        >
          
          TRANSLATE:public.header.news
        
        </a>
      </li>
    </ul>
  </div>
   
  <div
    class="flex"
  >
    <ul
      class="flex justify-end"
    >
      <li>
        <a
          class="my-auto text-white"
        >
          
          TRANSLATE:public.header.login
        
        </a>
      </li>
    </ul>
  </div>
</header>
`;
